### ISSUE TITLE
Checkboxes and relationships

### ISSUE THREAD

**Issue by agentphoenix** [LaravelCollective/html#23]
I'm using the FormBuilder and have a checkbox named the same thing as one of my model's relationships (a many-to-many relationship), but when using form model binding, it ignores any attempts I make to manually override the status of whether it's checked or not.

```php
{!! Form::checkbox('tags[]', $id, (in_array($id, $articleTags)) !!} {{ $tag }}
```

Doing this, I would expect it to ignore the relationship piece (which is tries to handle, albeit unsuccessfully, behind the scenes) and instead, use the logic I put in there. Instead, every checkbox is checked, even if I put `false` in the third parameter.

Am I missing something about how to get this to work or have I stumbled across an issue?

**Comment by robgridley** [comment]
I was able to get form model binding working my belongs to many and many to many relationships by doing the following:

1. Eager load the relationships in my controller.
2. Extend FormBuilder and override one method:

```php
protected function getModelValueAttribute($name)
{
    $value = parent::getModelValueAttribute($name);

    return ($value instanceof Collection) ? $value->modelKeys() : $value;
}
```

**Comment by wiesson** [comment]
@robgridley where do I have to put your extend code?

**Comment by voydz** [comment]
:+1: thanks for the solution @robgridley , I expect the same behavior as mentioned by @agentphoenix

**Comment by agentphoenix** [comment]
@adamgoose will this fix (or a similar fix) be merged into the repository or is the expectation that individuals will need to extend the `getModelValueAttribute` method ourselves to get form model binding working?

**Comment by adamgoose** [comment]
Yes, it was fixed in #45

**Comment by agentphoenix** [comment]
Awesome! Thanks for fixing this. Huge help! :+1:
